fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -475,13 +475,26 @@ struct PhysicalShard {
 	}
 
 	~PhysicalShard() {
-		if (!deletePending)
+		logShardEvent(id, ShardOp::CLOSE);
+		isInitialized.store(false);
+		readIterPool.reset();
+
+		// Deleting default column family is not allowed.
+		if (id == "default") {
 			return;
+		}
 
-		// Destroy CF
-		auto s = db->DropColumnFamily(cf);
+		if (deletePending) {
+			auto s = db->DropColumnFamily(cf);
```

Comment:
Could there be some retryable errors, like IO-timeout?

Reply:
Shouldn't be. The drop here only marks the CF for deletion. The actual delete happens after all the resources are released.